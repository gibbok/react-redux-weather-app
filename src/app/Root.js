// @flow
import React from 'react'
import { Provider } from 'react-redux'
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom'
import Forecast from './Forecast'
import Map from './Map'
import { withStyles } from 'material-ui'

const styles = {
  '@global': {
    body: {
      margin: 0,
      padding: 0
    },
    a: {
      textDecoration: 'none'
    }
  }
}

const Root = ({ store } : { store : Object }) => (
  <Provider store={store}>
    <Router>
      <Switch>
        <Route exact path='/' component={Forecast} />
        <Route exact path='/map' component={Map} />
      </Switch>
    </Router>
  </Provider>
)

export default withStyles(styles)(Root)
