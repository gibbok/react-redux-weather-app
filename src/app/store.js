import { createStore, applyMiddleware } from 'redux'
import thunkMiddleware from 'redux-thunk'
import promiseMiddleware from 'redux-promise-middleware'
import reducers from './reducers'

const store = createStore(reducers, applyMiddleware(
  thunkMiddleware,
  promiseMiddleware()
))

export default store
