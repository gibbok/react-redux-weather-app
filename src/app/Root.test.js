import React from 'react'
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json'
import Root from './Root'
import store from './store'

describe('<Root />', () => {
  it('should render', () => {
    const wrapper = shallow(<Root store={store} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
