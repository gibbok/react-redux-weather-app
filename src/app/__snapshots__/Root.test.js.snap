// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Root /> should render 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/map"
      />
    </Switch>
  </BrowserRouter>
</Provider>
`;
